src/crypto/crypto.js

```diff
@@ -264,33 +372,50 @@ export default {
    * @async
    */
   generateParams: function(algo, bits, oid) {
-    const types = [].concat(this.getPubKeyParamTypes(algo), this.getPrivKeyParamTypes(algo));
+    const paramNames = [].concat(this.getPublicKeyParamNames(algo), this.getPrivateKeyParamNames(algo));
     switch (algo) {
       case enums.publicKey.rsaEncrypt:
       case enums.publicKey.rsaEncryptSign:
       case enums.publicKey.rsaSign: {
         return publicKey.rsa.generate(bits, "10001").then(function(keyObject) {
-          return constructParams(
-            types, [keyObject.n, keyObject.e, keyObject.d, keyObject.p, keyObject.q, keyObject.u]
-          );
+          const { n, e, d, p, q, u } = keyObject;
+          const params = { n, e, d, p, q, u };
+          paramNames.forEach(name => { params[name] = params[name].toArrayLike(Uint8Array); });
+          return params;
```

Comment:
I would modify `publicKey.rsa.generate` to return `Uint8Array`s instead, and then make this a one-liner function call, if possible. And similarly below.

Reply:
The ECC generation functions return BN as well, I would keep that consistent, with BigInteger the conversion to Uint8Array will look nicer anyway